### PR TITLE
Added edit endpoint

### PR DESCRIPTION
**Pull request by VLiamin**

### CODE REVIEW

File: src/ProjectService.Business/Commands/File/EditFilesCommand.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _responseCreator = responseCreator;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid fileId, FileAccessType accessType)
+    {
+      DbProjectFile file = (await _repository.GetAsync(new List<Guid>() { fileId })).FirstOrDefault();
+      if (file is null
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/File/EditFilesCommand.cs:44]: if file is null => not found (404)

File: src/ProjectService.Business/Commands/File/EditFilesCommand.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      OperationResultResponse<bool> response;
+      if (accessType != (FileAccessType)file.Access)
+      {
+        response = new(
+          body: await _repository.EditAsync(fileId, accessType));
+      } 
+      else
+      {
+        response = new(body: true);
+      }
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/File/EditFilesCommand.cs:60]: response.body = accessType != (FileAccessType)file.Access ? body: await _repository.EditAsync(fileId, accessType)); : true;

**Review by MariyaSin — APPROVED** [review]

File: src/ProjectService.Business/Commands/File/EditFilesCommand.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      _repository = repository;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _responseCreator = responseCreator;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid fileId, FileAccessType accessType)
+    {
+      DbProjectFile file = (await _repository.GetAsync(new List<Guid>() { fileId })).FirstOrDefault();
```
Comment (ndoriomedov) [inline src/ProjectService.Business/Commands/File/EditFilesCommand.cs:43]: add empty line after this

File: src/ProjectService.Data/ProjectFileRepository.cs [hunk 4]
```diff
@@ -69,6 +69,21 @@ public async Task<bool> RemoveAsync(List<Guid> filesIds)
       await _provider.SaveAsync();
 
       return true;
+    }
+
+    public async Task<bool> EditAsync(Guid fileId, FileAccessType access)
+    {
+      DbProjectFile dbFile = await _provider.ProjectsFiles.FirstOrDefaultAsync(p => p.FileId == fileId);
```
Comment (ndoriomedov) [inline src/ProjectService.Data/ProjectFileRepository.cs:76]: same

**Review by ndoriomedov — APPROVED** [review]

File: src/ProjectService.Business/Commands/File/EditFilesCommand.cs (hunk 1 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/File/EditFilesCommand.cs]: done

File: src/ProjectService.Business/Commands/File/EditFilesCommand.cs (hunk 2 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/File/EditFilesCommand.cs:53]: done

File: src/ProjectService.Data/ProjectFileRepository.cs (hunk 4 above)
Comment (VLiamin) [reply src/ProjectService.Data/ProjectFileRepository.cs:76]: done

File: src/ProjectService.Business/Commands/File/EditFilesCommand.cs (hunk 3 above)
Comment (VLiamin) [reply src/ProjectService.Business/Commands/File/EditFilesCommand.cs:43]: done

**Pull request merged**